Added frimacs recipe.

pdo opened the pull request:
### Brief summary of what the package does

Provides a suite of major modes for working with the FriCAS computer algebra system.

### Direct link to the package repository

https://github.com/pdo/frimacs

### Your association with the package

Author and maintainer.

### Relevant communications with the upstream package maintainer

None needed.

### Checklist

<!-- Please confirm with `x`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this!  Very quick first pass below.

A couple things at the outset: first, I realize there is a lot here; feel free to ask if anything is unclear; the other is that I realize some of the .el files are meant to be data files.  It could help to not give them the elisp extension (e.g. name them .dat), which'll make it clear to the tooling I'm using that these aren't meant to be part of the code.

`fricas-standard-category-info.el` with byte-compile using Emacs 27.1:
- Temporarily ignoring `no-byte-compile` flag
```
fricas-standard-category-info.el:2:1:Warning: `("ABELGRP" . "AbelianGroup")'
    is a malformed function
```

`fricas-standard-category-info.el` with checkdoc 0.6.2:
```
fricas-standard-category-info.el:0: The first line should be of the form: ";;; package --- Summary"
fricas-standard-category-info.el:0: You should have a section marked ";;; Commentary:"
fricas-standard-category-info.el:2: You should have a section marked ";;; Code:"
fricas-standard-category-info.el:266: The footer should be: (provide 'fricas-standard-category-info)\n;;; fricas-standard-category-info.el ends here
```

`fricas-standard-category-info.el` with package-lint 20220412.1648:
```
1 issue found:
1:0: error: There is no `provide' form.
```

`fricas-standard-domain-info.el` with byte-compile using Emacs 27.1:
- Temporarily ignoring `no-byte-compile` flag
```
fricas-standard-domain-info.el:2:1:Warning: `("ALGSC"
    . "AlgebraGivenByStructuralConstants")' is a malformed function
fricas-standard-domain-info.el:2:1:Warning: `nil' called as a function
fricas-standard-domain-info.el:2:1:Error: Wrong type argument: listp, "Enumeration"
```

`fricas-standard-domain-info.el` with checkdoc 0.6.2:
```
fricas-standard-domain-info.el:0: The first line should be of the form: ";;; package --- Summary"
fricas-standard-domain-info.el:0: You should have a section marked ";;; Commentary:"
fricas-standard-domain-info.el:2: You should have a section marked ";;; Code:"
fricas-standard-domain-info.el:417: The footer should be: (provide 'fricas-standard-domain-info)\n;;; fricas-standard-domain-info.el ends here
```

`fricas-standard-domain-info.el` with package-lint 20220412.1648:
```
1 issue found:
1:0: error: There is no `provide' form.
```

`fricas-standard-operation-info.el` with byte-compile using Emacs 27.1:
- Temporarily ignoring `no-byte-compile` flag
```
fricas-standard-operation-info.el:2:1:Warning: `"0"' is a malformed function
```

`fricas-standard-operation-info.el` with checkdoc 0.6.2:
```
fricas-standard-operation-info.el:0: The first line should be of the form: ";;; package --- Summary"
fricas-standard-operation-info.el:0: You should have a section marked ";;; Commentary:"
fricas-standard-operation-info.el:2: You should have a section marked ";;; Code:"
fricas-standard-operation-info.el:2: The footer should be: (provide 'fricas-standard-operation-info)\n;;; fricas-standard-operation-info.el ends here
```

`fricas-standard-operation-info.el` with package-lint 20220412.1648:
```
1 issue found:
1:0: error: There is no `provide' form.
```

`fricas-standard-package-info.el` with byte-compile using Emacs 27.1:
- Temporarily ignoring `no-byte-compile` flag
```
fricas-standard-package-info.el:2:1:Warning: `("AF" . "AlgebraicFunction")' is
    a malformed function
```

`fricas-standard-package-info.el` with checkdoc 0.6.2:
```
fricas-standard-package-info.el:0: The first line should be of the form: ";;; package --- Summary"
fricas-standard-package-info.el:0: You should have a section marked ";;; Commentary:"
fricas-standard-package-info.el:2: You should have a section marked ";;; Code:"
fricas-standard-package-info.el:547: The footer should be: (provide 'fricas-standard-package-info)\n;;; fricas-standard-package-info.el ends here
```

`fricas-standard-package-info.el` with package-lint 20220412.1648:
```
1 issue found:
1:0: error: There is no `provide' form.
```

`frimacs-base.el` with checkdoc 0.6.2:
```
frimacs-base.el:0: The first line should be of the form: ";;; package --- Summary"
frimacs-base.el:0: You should have a section marked ";;; Commentary:"
frimacs-base.el:18: You should have a section marked ";;; Code:"
frimacs-base.el:508: The footer should be: (provide 'frimacs-base)\n;;; frimacs-base.el ends here
frimacs-base.el:325: All variables and subroutines might as well have a documentation string
frimacs-base.el:362: All interactive functions should have documentation
frimacs-base.el:494: All variables and subroutines might as well have a documentation string
frimacs-base.el:504: All variables and subroutines might as well have a documentation string
```

`frimacs-base.el` with package-lint 20220412.1648:
```
3 issues found:
4:3: warning: You should include standard keywords: see the variable `finder-known-keywords'.
458:16: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-incf'.
481:16: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-incf'.
```

More feedback on `frimacs-base.el` from [melpazoid](https://github.com/riscy/melpazoid):
```
- frimacs-base.el#L115: It's safer to sharp-quote function names; use `#'`
- frimacs-base.el#L119: It's safer to sharp-quote function names; use `#'`
- frimacs-base.el#L123: It's safer to sharp-quote function names; use `#'`
- frimacs-base.el#L127: It's safer to sharp-quote function names; use `#'`
- frimacs-base.el#L142: It's safer to sharp-quote function names; use `#'`
- frimacs-base.el#L146: It's safer to sharp-quote function names; use `#'`
- frimacs-base.el#L150: It's safer to sharp-quote function names; use `#'`
- frimacs-base.el#L154: It's safer to sharp-quote function names; use `#'`
- frimacs-base.el#L169: It's safer to sharp-quote function names; use `#'`
- frimacs-base.el#L173: It's safer to sharp-quote function names; use `#'`
- frimacs-base.el#L177: It's safer to sharp-quote function names; use `#'`
- frimacs-base.el#L181: It's safer to sharp-quote function names; use `#'`
- frimacs-base.el#L200: It's safer to sharp-quote function names; use `#'`
```

`frimacs-boot-mode.el` with byte-compile using Emacs 27.1:
```
frimacs-boot-mode.el:16:1:Error: Opening input file: No such file or directory, /workspace/pkg/data/fricas-standard-package-info.el
```

`frimacs-boot-mode.el` with checkdoc 0.6.2:
```
frimacs-boot-mode.el:0: The first line should be of the form: ";;; package --- Summary"
frimacs-boot-mode.el:0: You should have a section marked ";;; Commentary:"
frimacs-boot-mode.el:18: You should have a section marked ";;; Code:"
frimacs-boot-mode.el:100: The footer should be: (provide 'frimacs-boot-mode)\n;;; frimacs-boot-mode.el ends here
```

`frimacs-boot-mode.el` with package-lint 20220412.1648:
```
1 issue found:
4:3: warning: You should include standard keywords: see the variable `finder-known-keywords'.
```

`frimacs-buffer-menu.el` with checkdoc 0.6.2:
```
frimacs-buffer-menu.el:0: The first line should be of the form: ";;; package --- Summary"
frimacs-buffer-menu.el:0: You should have a section marked ";;; Commentary:"
frimacs-buffer-menu.el:13: You should have a section marked ";;; Code:"
frimacs-buffer-menu.el:301: The footer should be: (provide 'frimacs-buffer-menu)\n;;; frimacs-buffer-menu.el ends here
```

`frimacs-buffer-menu.el` with package-lint 20220412.1648:
```
1 issue found:
4:3: warning: You should include standard keywords: see the variable `finder-known-keywords'.
```

`frimacs-company.el` with byte-compile using Emacs 27.1:
```
frimacs-company.el:37:13:Warning: reference to free variable
    `frimacs-standard-names'
frimacs-company.el:38:11:Warning: reference to free variable
    `frimacs-standard-names-and-abbreviations'
frimacs-company.el:45:32:Warning: reference to free variable
    `frimacs-process-repl-buffer-name'
frimacs-company.el:56:17:Warning: reference to free variable
    `company-backends'
frimacs-company.el:56:17:Warning: assignment to free variable
    `company-backends'
In end of data:
frimacs-company.el:59:1:Warning: the following functions are not known to be
    defined: company-begin-backend, company-grab-symbol,
    frimacs-process-constructor-type, frimacs-process-verify-operation-name,
    frimacs-process-document-operation,
    frimacs-process-verify-constructor-name-or-abbrev,
    frimacs-process-document-constructor,
    frimacs-process-find-constructor-source
```

`frimacs-company.el` with checkdoc 0.6.2:
```
frimacs-company.el:0: The first line should be of the form: ";;; package --- Summary"
frimacs-company.el:0: You should have a section marked ";;; Commentary:"
frimacs-company.el:16: You should have a section marked ";;; Code:"
frimacs-company.el:58: The footer should be: (provide 'frimacs-company)\n;;; frimacs-company.el ends here
```

`frimacs-company.el` with package-lint 20220412.1648:
```
9 issues found:
4:3: warning: You should include standard keywords: see the variable `finder-known-keywords'.
14:10: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-lib'.
16:1: error: You should depend on (emacs "24.4") if you need `with-eval-after-load'.
16:1: warning: `with-eval-after-load' is for use in configurations, and should rarely be used in packages.
27:5: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-case'.
34:8: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-remove-if-not'.
40:8: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-case'.
54:18: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-first'.
54:38: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-second'.
```

`frimacs-help-mode.el` with byte-compile using Emacs 27.1:
```
frimacs-help-mode.el:13:1:Error: Opening input file: No such file or directory, /workspace/pkg/data/fricas-standard-package-info.el
```

`frimacs-help-mode.el` with checkdoc 0.6.2:
```
frimacs-help-mode.el:0: The first line should be of the form: ";;; package --- Summary"
frimacs-help-mode.el:0: You should have a section marked ";;; Commentary:"
frimacs-help-mode.el:15: You should have a section marked ";;; Code:"
frimacs-help-mode.el:53: The footer should be: (provide 'frimacs-help-mode)\n;;; frimacs-help-mode.el ends here
```

`frimacs-help-mode.el` with package-lint 20220412.1648:
```
1 issue found:
4:3: warning: You should include standard keywords: see the variable `finder-known-keywords'.
```

`frimacs-input-mode.el` with byte-compile using Emacs 27.1:
```
frimacs-input-mode.el:13:1:Error: Opening input file: No such file or directory, /workspace/pkg/data/fricas-standard-package-info.el
```

`frimacs-input-mode.el` with checkdoc 0.6.2:
```
frimacs-input-mode.el:0: The first line should be of the form: ";;; package --- Summary"
frimacs-input-mode.el:0: You should have a section marked ";;; Commentary:"
frimacs-input-mode.el:18: You should have a section marked ";;; Code:"
frimacs-input-mode.el:175: The footer should be: (provide 'frimacs-input-mode)\n;;; frimacs-input-mode.el ends here
```

`frimacs-input-mode.el` with package-lint 20220412.1648:
```
1 issue found:
4:3: warning: You should include standard keywords: see the variable `finder-known-keywords'.
```

More feedback on `frimacs-input-mode.el` from [melpazoid](https://github.com/riscy/melpazoid):
```
- frimacs-input-mode.el#L161: It's safer to sharp-quote function names; use `#'`
```

`frimacs-process-mode.el` with byte-compile using Emacs 27.1:
```
frimacs-process-mode.el:14:1:Error: Opening input file: No such file or directory, /workspace/pkg/data/fricas-standard-package-info.el
```

`frimacs-process-mode.el` with checkdoc 0.6.2:
```
frimacs-process-mode.el:0: The first line should be of the form: ";;; package --- Summary"
frimacs-process-mode.el:0: You should have a section marked ";;; Commentary:"
frimacs-process-mode.el:18: You should have a section marked ";;; Code:"
frimacs-process-mode.el:1187: The footer should be: (provide 'frimacs-process-mode)\n;;; frimacs-process-mode.el ends here
```

`frimacs-process-mode.el` with package-lint 20220412.1648:
```
23 issues found:
4:3: warning: You should include standard keywords: see the variable `finder-known-keywords'.
13:10: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-lib'.
123:1: error: You should depend on (emacs "24.3") if you need `defvar-local'.
125:1: error: You should depend on (emacs "24.3") if you need `defvar-local'.
127:1: error: You should depend on (emacs "24.3") if you need `defvar-local'.
129:1: error: You should depend on (emacs "24.3") if you need `defvar-local'.
564:25: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-subseq'.
822:24: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-first'.
825:25: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-second'.
933:5: error: You should depend on (emacs "26.1") if you need `indent-relative-first-indent-point'.
1037:13: error: You should depend on (emacs "25.1") if you need `make-process'.
1074:30: error: You should depend on (emacs "24.4") if you need `string-trim'.
1075:27: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-find-if'.
1081:43: error: You should depend on (emacs "26.1") if you need `file-attribute-size'.
1095:15: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-pushnew'.
1100:23: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-remove-if'.
1103:17: error: You should depend on (emacs "25.1") if you need `make-process'.
1138:30: error: You should depend on (emacs "24.4") if you need `string-trim'.
1139:27: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-find-if'.
1145:43: error: You should depend on (emacs "26.1") if you need `file-attribute-size'.
1155:15: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-pushnew'.
1160:23: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-remove-if'.
1173:0: error: "run-fricas" doesn't start with package's prefix "frimacs".
```

More feedback on `frimacs-process-mode.el` from [melpazoid](https://github.com/riscy/melpazoid):
```
- frimacs-process-mode.el#L963: It's safer to sharp-quote function names; use `#'`
- frimacs-process-mode.el#L241: Could this point/point-max comparison use `eobp`?
- frimacs-process-mode.el#L245: Could this point/point-min comparison use `bobp`?
- frimacs-process-mode.el#L305: No `format` required; `message` takes an f-string
```

`frimacs-selector.el` with byte-compile using Emacs 27.1:
```
frimacs-selector.el:86:26:Warning: reference to free variable
    `frimacs-process-repl-buffer-name'
In end of data:
frimacs-selector.el:122:1:Warning: the function `frimacs-buffer-menu' is not
    known to be defined.
```

`frimacs-selector.el` with checkdoc 0.6.2:
```
frimacs-selector.el:0: The first line should be of the form: ";;; package --- Summary"
frimacs-selector.el:0: You should have a section marked ";;; Commentary:"
frimacs-selector.el:17: You should have a section marked ";;; Code:"
frimacs-selector.el:121: The footer should be: (provide 'frimacs-selector)\n;;; frimacs-selector.el ends here
```

`frimacs-selector.el` with package-lint 20220412.1648:
```
7 issues found:
4:3: warning: You should include standard keywords: see the variable `finder-known-keywords'.
15:10: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-lib'.
50:21: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-third'.
71:35: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-first'.
71:52: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-second'.
97:30: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-first'.
100:18: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-rest'.
```

`frimacs-spad-mode.el` with byte-compile using Emacs 27.1:
```
frimacs-spad-mode.el:14:1:Error: Opening input file: No such file or directory, /workspace/pkg/data/fricas-standard-package-info.el
```

`frimacs-spad-mode.el` with checkdoc 0.6.2:
```
frimacs-spad-mode.el:0: The first line should be of the form: ";;; package --- Summary"
frimacs-spad-mode.el:0: You should have a section marked ";;; Commentary:"
frimacs-spad-mode.el:18: You should have a section marked ";;; Code:"
frimacs-spad-mode.el:150: The footer should be: (provide 'frimacs-spad-mode)\n;;; frimacs-spad-mode.el ends here
```

`frimacs-spad-mode.el` with package-lint 20220412.1648:
```
1 issue found:
4:3: warning: You should include standard keywords: see the variable `finder-known-keywords'.
```

More feedback on `frimacs-spad-mode.el` from [melpazoid](https://github.com/riscy/melpazoid):
```
- frimacs-spad-mode.el#L137: It's safer to sharp-quote function names; use `#'`
```

`frimacs.el` with byte-compile using Emacs 27.1:
```
frimacs.el:56:1:Error: Opening input file: No such file or directory, /workspace/pkg/data/fricas-standard-package-info.el
```

`frimacs.el` with package-lint 20220412.1648:
```
1 issue found:
1:0: warning: "Version:" or "Package-Version:" header is missing. MELPA will handle this, but other archives will not.
```

`ob-autoload.el` with byte-compile using Emacs 27.1:
```
ob-autoload.el:2:1:Error: Opening input file: No such file or directory, /workspace/pkg/data/fricas-standard-package-info.el
```

`ob-autoload.el` with checkdoc 0.6.2:
```
ob-autoload.el:0: The first line should be of the form: ";;; package --- Summary"
ob-autoload.el:0: You should have a section marked ";;; Commentary:"
ob-autoload.el:2: You should have a section marked ";;; Code:"
ob-autoload.el:2: The footer should be: (provide 'ob-autoload)\n;;; ob-autoload.el ends here
```

`ob-autoload.el` with package-lint 20220412.1648:
```
1 issue found:
1:0: error: There is no `provide' form.
```

`ob-fricas.el` with byte-compile using Emacs 27.1:
```
ob-fricas.el:73:1:Error: Opening input file: No such file or directory, /workspace/pkg/data/fricas-standard-package-info.el
```

`ob-fricas.el` with checkdoc 0.6.2:
```
ob-fricas.el:0: The first line should be of the form: ";;; package --- Summary"
ob-fricas.el:0: You should have a section marked ";;; Commentary:"
ob-fricas.el:78: You should have a section marked ";;; Code:"
ob-fricas.el:254: The footer should be: (provide 'ob-fricas)\n;;; ob-fricas.el ends here
```

`ob-fricas.el` with package-lint 20220412.1648:
```
16 issues found:
4:3: warning: You should include standard keywords: see the variable `finder-known-keywords'.
71:10: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-lib'.
78:0: error: "org-babel-header-args:fricas" doesn't start with package's prefix "frimacs".
78:0: error: `org-babel-header-args:fricas' contains a non-standard separator `:', use hyphens instead (see Elisp Coding Conventions).
99:0: error: "org-babel-fricas--starify-name" doesn't start with package's prefix "frimacs".
110:0: error: "org-babel-fricas-initiate-session" doesn't start with package's prefix "frimacs".
127:0: error: "org-babel-fricas-var-to-fricas" doesn't start with package's prefix "frimacs".
137:15: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-mapcan'.
146:0: error: "org-babel-expand-body:fricas" doesn't start with package's prefix "frimacs".
146:0: error: `org-babel-expand-body:fricas' contains a non-standard separator `:', use hyphens instead (see Elisp Coding Conventions).
168:0: error: "org-babel-fricas--body-needs-block-read" doesn't start with package's prefix "frimacs".
172:0: error: "org-babel-fricas--execute-by-block-read" doesn't start with package's prefix "frimacs".
204:0: error: "org-babel-fricas--execute-line-by-line" doesn't start with package's prefix "frimacs".
239:0: error: "org-babel-header-args:axiom" doesn't start with package's prefix "frimacs".
239:0: error: `org-babel-header-args:axiom' contains a non-standard separator `:', use hyphens instead (see Elisp Coding Conventions).
251:0: error: Aliases should start with the package's prefix "frimacs".
```

More feedback on `ob-fricas.el` from [melpazoid](https://github.com/riscy/melpazoid):
```
- ob-fricas.el#L250: It's safer to sharp-quote function names; use `#'`
- ob-fricas.el#L251: It's safer to sharp-quote function names; use `#'`
- ob-fricas.el#L252: It's safer to sharp-quote function names; use `#'`
```

`load`-check on each file:
```
Loading fricas-standard-category-info.el
  fricas-standard-category-info.el:Error: Emacs 27.1 couldn't load:
  (invalid-function ("ABELGRP" . "AbelianGroup"))
Loading fricas-standard-domain-info.el
  fricas-standard-domain-info.el:Error: Emacs 27.1 couldn't load:
  (invalid-function ("ALGSC" . "AlgebraGivenByStructuralConstants"))
Loading fricas-standard-operation-info.el
  fricas-standard-operation-info.el:Error: Emacs 27.1 couldn't load:
  (invalid-function "0")
Loading fricas-standard-package-info.el
  fricas-standard-package-info.el:Error: Emacs 27.1 couldn't load:
  (invalid-function ("AF" . "AlgebraicFunction"))
Loading frimacs-base.el
  frimacs-base.el:Error: Emacs 27.1 couldn't load:
  (file-missing "Opening input file" "No such file or directory" "/workspace/pkg/data/fricas-standard-package-info.el")
Loading frimacs-boot-mode.el
  frimacs-boot-mode.el:Error: Emacs 27.1 couldn't load:
  (file-missing "Opening input file" "No such file or directory" "/workspace/pkg/data/fricas-standard-package-info.el")
Loading frimacs-buffer-menu.el
Loading frimacs-company.el
Loading frimacs-help-mode.el
  frimacs-help-mode.el:Error: Emacs 27.1 couldn't load:
  (file-missing "Opening input file" "No such file or directory" "/workspace/pkg/data/fricas-standard-package-info.el")
Loading frimacs-input-mode.el
  frimacs-input-mode.el:Error: Emacs 27.1 couldn't load:
  (file-missing "Opening input file" "No such file or directory" "/workspace/pkg/data/fricas-standard-package-info.el")
Loading frimacs-process-mode.el
  frimacs-process-mode.el:Error: Emacs 27.1 couldn't load:
  (file-missing "Opening input file" "No such file or directory" "/workspace/pkg/data/fricas-standard-package-info.el")
Loading frimacs-selector.el
Loading frimacs-spad-mode.el
  frimacs-spad-mode.el:Error: Emacs 27.1 couldn't load:
  (file-missing "Opening input file" "No such file or directory" "/workspace/pkg/data/fricas-standard-package-info.el")
Loading frimacs.el
  frimacs.el:Error: Emacs 27.1 couldn't load:
  (file-missing "Opening input file" "No such file or directory" "/workspace/pkg/data/fricas-standard-package-info.el")
Loading ob-autoload.el
  ob-autoload.el:Error: Emacs 27.1 couldn't load:
  (file-missing "Opening input file" "No such file or directory" "/workspace/pkg/data/fricas-standard-package-info.el")
Loading ob-fricas.el
  ob-fricas.el:Error: Emacs 27.1 couldn't load:
  (file-missing "Opening input file" "No such file or directory" "/workspace/pkg/data/fricas-standard-package-info.el")
```

Stderr output while compiling/loading:
```
Loading standard FriCAS package information
Loading standard FriCAS package information
Loading standard FriCAS package information
Loading standard FriCAS package information
Loading standard FriCAS package information
Loading standard FriCAS package information
Loading standard FriCAS package information
Loading standard FriCAS package information
Loading standard FriCAS package information
```

Package and license:
- Prefer the default recipe or `:defaults`, if possible.
- GitHub API found `MIT License`
- data/fricas-standard-category-info.el (unknown license) -- (no header)
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to data/fricas-standard-category-info.el
- data/fricas-standard-domain-info.el (unknown license) -- (no header)
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to data/fricas-standard-domain-info.el
- data/fricas-standard-operation-info.el (unknown license) -- (no header)
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to data/fricas-standard-operation-info.el
- data/fricas-standard-package-info.el (unknown license) -- (no header)
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to data/fricas-standard-package-info.el
- frimacs-base.el (unknown license) -- (no header)
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to frimacs-base.el
- frimacs-boot-mode.el (unknown license) -- (no header)
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to frimacs-boot-mode.el
- frimacs-buffer-menu.el (unknown license) -- (no header)
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to frimacs-buffer-menu.el
- frimacs-company.el (unknown license) -- (no header)
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to frimacs-company.el
- frimacs-help-mode.el (unknown license) -- (no header)
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to frimacs-help-mode.el
- frimacs-input-mode.el (unknown license) -- (no header)
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to frimacs-input-mode.el
- frimacs-process-mode.el (unknown license) -- (no header)
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to frimacs-process-mode.el
- frimacs-selector.el (unknown license) -- (no header)
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to frimacs-selector.el
- frimacs-spad-mode.el (unknown license) -- (no header)
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to frimacs-spad-mode.el
- frimacs.el (unknown license) -- An environment for the FriCAS computer algebra system
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to frimacs.el
- ob-autoload.el (unknown license) -- (no header)
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to ob-autoload.el
- ob-fricas.el (unknown license) -- (no header)
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to ob-fricas.el

pdo commented:
Hi Chris,

I'll rename the data files to a .dat extension as requested.

The checkdoc messages of the form:-
  frimacs-base.el:0: The first line should be of the form: ";;; package --- Summary"
  frimacs-base.el:0: You should have a section marked ";;; Commentary:"
  frimacs-base.el:18: You should have a section marked ";;; Code:"
  frimacs-base.el:508: The footer should be: (provide 'frimacs-base)\n;;; frimacs-base.el ends here
  ...
I've been ignoring because I thought only the docstrings were required to pass checkdoc.  Do I really need to add these comments to each file?  I particularly dislike the first because it obscures my use of -*- lexical-binding: t -*- which I believe must also be on the first line for correct operation.  Note that I have followed the requested comment format in frimacs.el, which I had hoped would be enough for the tools to pick up the package description, dependencies, etc.

I'll also update the Emacs version dependency to 26.1 and look more closely at the other messages.

Thanks,
Paul

riscy commented:
> Do I really need to add these comments to each file? I particularly dislike the first because it obscures my use of -- lexical-binding: t -- which I believe must also be on the first line for correct operation.

I think you're right that it's technically not necessary for creating index metadata, but your package would be very unusual if you didn't do this, and this would set a precedent that hasn't been set.

One solution to the obscured lexical-binding line would be to make the Summary short -- e.g. "Part of frimacs  --- -*- lexical-binding: t -*-".

pdo commented:
> I think you're right that it's technically not necessary for creating index metadata, but your package would be very unusual if you didn't do this, and this would set a precedent that hasn't been set.

> One solution to the obscured lexical-binding line would be to make the Summary short -- e.g. "Part of frimacs --- -- lexical-binding: t --".

Okay, fair enough, I'll use a short (non-)description on the first line and put the structured comments in there to keep checkdoc happy.

Thanks,
Paul

pdo commented:
Hi Chris,

I've made some changes now which will hopefully cut out most of the noise in your tooling reports and allow us to see the important bits that are left.

If you could run them again so we can address any blockers that remain that would be great.

Thanks,
Paul

riscy commented:
Thank you @pdo!  The latest is below, but I wonder if the `ob-*` files belong with this recipe.  Maybe they should be removed/separated into another package?

`frimacs-boot-mode.el` with byte-compile using Emacs 27.1:
```
frimacs-boot-mode.el:170:1:Warning: the function `company-complete' is not
    known to be defined.
```

`frimacs-company.el` with byte-compile using Emacs 27.1:
```
frimacs-company.el:52:32:Warning: reference to free variable
    `frimacs-process-repl-buffer-name'
frimacs-company.el:63:17:Warning: reference to free variable
    `company-backends'
frimacs-company.el:63:17:Warning: assignment to free variable
    `company-backends'
In end of data:
frimacs-company.el:68:1:Warning: the following functions are not known to be
    defined: company-begin-backend, company-grab-symbol,
    frimacs-process-constructor-type, frimacs-process-verify-operation-name,
    frimacs-process-document-operation,
    frimacs-process-verify-constructor-name-or-abbrev,
    frimacs-process-document-constructor,
    frimacs-process-find-constructor-source
```

`frimacs-company.el` with package-lint 20220412.1648:
```
1 issue found:
23:1: warning: `with-eval-after-load' is for use in configurations, and should rarely be used in packages.
```

`frimacs-input-mode.el` with byte-compile using Emacs 27.1:
```
frimacs-input-mode.el:185:1:Warning: the function `company-complete' is not
    known to be defined.
```

`frimacs-process-mode.el` with byte-compile using Emacs 27.1:
```
frimacs-process-mode.el:1200:1:Warning: the function `company-complete' is not
    known to be defined.
```

`frimacs-process-mode.el` with package-lint 20220412.1648:
```
1 issue found:
1183:0: error: "run-fricas" doesn't start with package's prefix "frimacs".
```

More feedback on `frimacs-process-mode.el` from [melpazoid](https://github.com/riscy/melpazoid):
```
- frimacs-process-mode.el#L248: Could this point/point-max comparison use `eobp`?
- frimacs-process-mode.el#L252: Could this point/point-min comparison use `bobp`?
- frimacs-process-mode.el#L312: No `format` required; `message` takes an f-string
```

`frimacs-selector.el` with byte-compile using Emacs 27.1:
```
frimacs-selector.el:131:1:Warning: the function `frimacs-buffer-menu' is not
    known to be defined.
```

`frimacs-spad-mode.el` with byte-compile using Emacs 27.1:
```
frimacs-spad-mode.el:160:1:Warning: the function `company-complete' is not
    known to be defined.
```

`frimacs.el` with package-lint 20220412.1648:
```
2 issues found:
1:0: warning: "Version:" or "Package-Version:" header is missing. MELPA will handle this, but other archives will not.
1:0: warning: The package summary is too long. It should be at most 60 characters.
```

`ob-autoload.el` with package-lint 20220412.1648:
```
1 issue found:
1:0: error: There is no `provide' form.
```

`ob-fricas.el` with byte-compile using Emacs 27.1:
```
ob-fricas.el:117:1:Warning: Unused lexical argument `params'
```

`ob-fricas.el` with package-lint 20220412.1648:
```
13 issues found:
85:0: error: "org-babel-header-args:fricas" doesn't start with package's prefix "frimacs".
85:0: error: `org-babel-header-args:fricas' contains a non-standard separator `:', use hyphens instead (see Elisp Coding Conventions).
106:0: error: "org-babel-fricas--starify-name" doesn't start with package's prefix "frimacs".
117:0: error: "org-babel-fricas-initiate-session" doesn't start with package's prefix "frimacs".
134:0: error: "org-babel-fricas-var-to-fricas" doesn't start with package's prefix "frimacs".
153:0: error: "org-babel-expand-body:fricas" doesn't start with package's prefix "frimacs".
153:0: error: `org-babel-expand-body:fricas' contains a non-standard separator `:', use hyphens instead (see Elisp Coding Conventions).
175:0: error: "org-babel-fricas--body-needs-block-read" doesn't start with package's prefix "frimacs".
179:0: error: "org-babel-fricas--execute-by-block-read" doesn't start with package's prefix "frimacs".
211:0: error: "org-babel-fricas--execute-line-by-line" doesn't start with package's prefix "frimacs".
246:0: error: "org-babel-header-args:axiom" doesn't start with package's prefix "frimacs".
246:0: error: `org-babel-header-args:axiom' contains a non-standard separator `:', use hyphens instead (see Elisp Coding Conventions).
258:0: error: Aliases should start with the package's prefix "frimacs".
```

Package and license:
- frimacs-base.el (unknown license) -- Part of frimacs
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to frimacs-base.el
- frimacs-boot-mode.el (unknown license) -- Part of frimacs
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to frimacs-boot-mode.el
- frimacs-buffer-menu.el (unknown license) -- Part of frimacs
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to frimacs-buffer-menu.el
- frimacs-company.el (unknown license) -- Part of frimacs
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to frimacs-company.el
- frimacs-help-mode.el (unknown license) -- Part of frimacs
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to frimacs-help-mode.el
- frimacs-input-mode.el (unknown license) -- Part of frimacs
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to frimacs-input-mode.el
- frimacs-process-mode.el (unknown license) -- Part of frimacs
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to frimacs-process-mode.el
- frimacs-selector.el (unknown license) -- Part of frimacs
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to frimacs-selector.el
- frimacs-spad-mode.el (unknown license) -- Part of frimacs
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to frimacs-spad-mode.el
- frimacs.el (unknown license) -- An environment for working with the FriCAS computer algebra system
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to frimacs.el
- ob-autoload.el (unknown license) -- Part of frimacs
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to ob-autoload.el
- ob-fricas.el (unknown license) -- Part of frimacs
  - Add license boilerplate or an [SPDX-License-Identifier](https://spdx.dev/ids/) to ob-fricas.el

pdo commented:
Hi Chris,

>  The latest is below, but I wonder if the ob-* files belong with this recipe. Maybe they should be removed/separated into another package?

Okay, I can put the ob- files into another package (I'll call it "ob-fricas").

For the frimacs-company.el warnings, especially:-

> frimacs-company.el with package-lint 20220412.1648:
> 
> 1 issue found:
> 23:1: warning: `with-eval-after-load' is for use in configurations, and should rarely be used in packages.

I'm not sure what to do about them.  The intention was for the contents of that file to be evaluated only if company-mode is present, but I'm not sure I'm going about this the right way.  Is there an accepted mechanism for this kind of "optional dependency"?

Thanks,
Paul

pdo commented:
Hi Chris,

I've removed the frimacs-company.el and ob-*.el files from the package and also removed all references to company-complete from the remaining files.  I've added SPDX-License-Identifiers to all files and made a few other tweaks here and there.  I now believe (hope :-) all "objectionable" messages should be gone.

At some point I'll do recipes for frimacs-company and ob-fricas, but they'll be separate pull requests, so hopefully this pull request will now be able to go through.

Paul

riscy commented:
Thank you so much for accommodating, I realize it was a large laundry list.  Here are the last few nags --

`frimacs-process-mode.el` with package-lint 20220412.1648:
```
1 issue found:
1178:0: error: "run-fricas" doesn't start with package's prefix "frimacs".
```
The reason package-lint and I are so stodgy about these is to avoid other packages shadowing your functions.  You could recommend users run a `defalias` so that their "M-x" experience is simpler.

This one's worth addressing too:

`frimacs.el` with package-lint 20220412.1648:
```
1 issue found:
1:0: warning: "Version:" or "Package-Version:" header is missing. MELPA will handle this, but other archives will not.
```
Prefer `Version` in this case.  MELPA adds Package-Version.

Just hose two more should put the checklist behind us.

pdo commented:
Okay, I've renamed run-fricas to frimacs-run-fricas and added a Version: 1.0 header comment, so hopefully we're good to go now :-)

Paul

riscy commented:
Thank you for this!
